api/bases/dataplane.openstack.org_openstackdataplanenodesets.yaml

```diff
@@ -1062,6 +1071,827 @@ spec:
                       - subnetName
                       type: object
                     type: array
+                  nodes:
+                    additionalProperties:
+                      properties:
```

Comment:
so i think it might be useful to add a way to express group membership to the node specification

that would allow using group vars and groups in roles

That might reduce the number of dataplane services that are required as a result.

i.e. you coudl defien the nova role will execute on all nodes in the nova group. then tag some nodes with nova_east_coast adn other with nova_west_cost and defien nova as

[nova:children]
nova_east_coast
nova_west_cost

here the assumtion is that perhaps each group might be a diffent cell.
the altherinvie woudl be ot use spereate nodeset per cell/rejoin.
which is vlaid but then you also have to do that for other groupings.

ie. a node set for the 20 hosts with gpus and a diffent one for the sriov host and a diffent one for gpus and sriov.

this is one of the usecases i would hope we coudl adress in this propsoal

how to group host that may have multipel diffent roels./usecases.
im not sure what the right approch is but i wanted to raise this as an option to think about.